### PR TITLE
Add h index

### PR DESCRIPTION
**Pull request by alexpantyukhin**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by a-r-r-o-w — APPROVED** [review]
Looks great!

**Review by realstealthninja — APPROVED** [review]
LGTM

File: other/h_index.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    ValueError: The citations should be a list of non negative integers.
+    >>> h_index([1,2,-3])
+    Traceback (most recent call last):
+        ...
+    ValueError: The citations should be a list of non negative integers.
+    """
+
+    # validate:
+    if not isinstance(citations, list) or not all(
+        [isinstance(item, int) and item >= 0 for item in citations]
```
Comment (CaedenPH) [inline other/h_index.py:54]:
```suggestion
        isinstance(item, int) and item >= 0 for item in citations
```
This doesn't need to be a list comprehension, a generator would be faster in this case

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Small change, but other than that looks good!

File: other/h_index.py (hunk 1 above)
Comment (alexpantyukhin) [reply other/h_index.py]: thank you! I didn't know about that.

**Review by CaedenPH — APPROVED** [review]
Looks good

**Comment by alexpantyukhin** [comment]
Hello, what about this PR?

**Review by ChrisO345 — APPROVED** [review]

**Pull request merged**